src/main/java/uk/gov/justice/hmpps/casenotes/sync/SyncCaseNotes.kt

```diff
@@ -42,11 +39,12 @@ class SyncCaseNotes(
     }
 
     existing?.also {
-      check(it.prisonNumber == request.personIdentifier) { "Case note belongs to another prisoner or prisoner records have been merged" }
+      check(it.personIdentifier == request.personIdentifier) { "Case note belongs to another prisoner or prisoner records have been merged" }
+      noteRepository.delete(it)
```

Comment:
Possibly worth tracking a custom event for these updates via delete and recreate. Perhaps as part of a ticket to consolidate custom events